### ISSUE TITLE
How To: Log Message String Replacement

### ISSUE THREAD

**Issue by codepwner** [verticalsoftware/vertical-spectreconsolelogger#30]
What is the best way to configure or extend this library such that it performs a string replacement on every message logged?

For my use case, I have an application which runs in a cloud provider and they consume the console log messages into a log exploring system, ElasticSearch.  When a log message includes newlines, this results in many log items that are much harder to later review.  As an example, a standard C# exception logged to console will include a newline for each item in the stack trace.  If I were to need to search for a particular exception, I'd need to do a lot of work to correlate all the individual logged items together using some timestamp or similar.

In the built-in [SimpleConsoleLogger](https://learn.microsoft.com/en-us/dotnet/core/extensions/console-log-formatter#simple) provider, there is an option for my use case with a boolean property for SingleLine made available in the builder config.  I didn't see a similar option in this library, but wondered if there was a recommended way to include into the rendering pipeline a call to string.Replace where I could perform the substitution myself.  In my case, I would be replacing the newline character with U+2028.

**Comment by daningalla** [comment]
@codepwner  - ~~there is currently no facility to do this in the API~~. The complication is not so much the message template string, but the arguments as they are expanded and injected into the output. There would be a lot of "hook" points needed to accommodate that. Let me take a look though and see what the possibilities are. If there are any feasible ways, I'll create a new enhancement issue and tag you in it so we can discuss. Or, if you dig into the code and see a way please do the same.

Thank you for the feedback-always trying to make it better.

**Comment by daningalla** [comment]
@codepwner  - OK, there kind of is, see the sample [here](https://github.com/verticalsoftware/vertical-spectreconsolelogger/tree/for-codepwner/examples/CustomConsoleWriter). String replace is certainly not optimal, but if you need something quick this would work.

**Comment by codepwner** [comment]
Thank you for the guidance!!! I was able to get things working.  For anyone finding this later on with the same use case as me, I only made one deviation from the example...

It appears that Spectre needs a new line included, else several logs will overwrite one another (at least in consoles that support it).  I simply added back a newline at the end of the stripped content,

From
https://github.com/verticalsoftware/vertical-spectreconsolelogger/blob/5afdc5af300ca264caa27d84280ebd65bcdbbf8f/examples/CustomConsoleWriter/NewLineReplacingWriter.cs#L18-L20

To
`var replaced = content.Replace(Environment.NewLine, "\u2028") + Environment.NewLine;`

For the moment, this simple solution addresses my need.  Would be interested in adopting any better solutions down the road that aren't reliant on direct string manipulation, but this is a great leap forward overall.

@daningalla - I'd have marked this closed, but I wasn't sure if you wanted to keep it open if you wanted to keep it as a possible enhancement / explore other solutions.  If not, feel free to close it 👍🏽
